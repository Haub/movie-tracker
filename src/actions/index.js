export const displayHanksMovies = (movies) => ({
  type: 'DISPLAY_HANKS_MOVIES',
  movies
})

export const updateFavorites = (movies) => ({
  type: 'UPDATE_FAVORITES',
  movies
export const addFavorite = (movie) => ({
  type: 'ADD_FAVORITE',
  movie
});

export const removeFavorite = (movieId) => ({
  type: 'REMOVE_FAVORITE',
  movieId
});

export const loginUser = (user) => ({
  type: 'LOGIN_USER',
  user
});

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
});
