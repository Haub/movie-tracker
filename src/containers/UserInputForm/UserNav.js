import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import './Navigation.css';
import { logoutUser, updateFavorites } from '../../actions';

export const UserNav = ({ user, clearFavorites, logout }) => {

  const handleLogOut = () => {
    clearFavorites([]);
    logout();
  }

  return (
    <div className="user-nav">
      <p className="tooltip" data-tooltip="Select your favorite Tom Hanks movies!">?</p>
      <h3>{user.name}</h3>
      <Link to="/favorites" className="favorites btn">
        Favorites â˜…
      </Link>
      <Link to="/user" className="discover btn">
        Discover Hanks Movies
      </Link>
      <Link to="/" replace>
        <button className="log-out" onClick={handleLogOut}>Log Out</button>
      </Link>
    </div>
  )
}

export const mapStateToProps = ({ user }) => ({ user });

export const mapDispatchToProps = (dispatch) => ({
  logout: () => dispatch(logoutUser()),
  clearFavorites: (movies) => dispatch(updateFavorites(movies))
})

export default connect(mapStateToProps, mapDispatchToProps)(UserNav)