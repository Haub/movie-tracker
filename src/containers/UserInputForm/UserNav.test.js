import React from 'react';
import { shallow } from 'enzyme';
import { UserNav, mapDispatchToProps, mapStateToProps } from './UserNav';
import { logoutUser, updateFavorites } from '../../actions';

describe('UserNav Component', () => {
  let wrapper;
  let user; 
  let clearFavorites; 
  let logout;

  beforeEach(() => {
    logout = jest.fn(); 
    clearFavorites = jest.fn();
    user = { name:'tim' };

    wrapper = shallow(
      <UserNav 
        logout={logout}
        clearFavorites={clearFavorites}
        user={user}
      />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should logout user and clear favorites on Log Out button click', () => {
    wrapper.find('button').simulate('click');
    expect(logout).toHaveBeenCalled();
    expect(clearFavorites).toHaveBeenCalled();
  });

  describe('mapStateToProps', () => {
    it('should return a user object', () => {
      const name = 'Tim';
      const id = 2;
      const email = 'foo@barr';
      const password = 'oops';
      const mockStore = {
        user: {name, id, email, password},
        movies: [],
        favorites: []
      };
      const expected = {
        user: {name, id, email, password}
      };
      const result = mapStateToProps(mockStore);
      expect(result).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should log out a user', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = logoutUser();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.logout();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should clear user\'s favorites on log out', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = updateFavorites([]);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.clearFavorites([]);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

});
