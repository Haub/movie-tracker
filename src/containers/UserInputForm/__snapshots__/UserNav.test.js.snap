// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNav Component should match snapshot 1`] = `
<div
  className="NavBar"
>
  <p
    className="tooltip"
    data-tooltip="Select your favorite Tom Hanks movies!"
  >
    ?
  </p>
  <h3>
    tim
  </h3>
  <Link
    className="favorites btn"
    replace={false}
    to="/favorites"
  >
    Favorites â˜…
  </Link>
  <Link
    className="discover btn"
    replace={false}
    to="/user"
  >
    Discover Hanks Movies
  </Link>
  <Link
    replace={true}
    to="/"
  >
    <button
      onClick={[Function]}
    >
      Log Out
    </button>
  </Link>
</div>
`;
